crates/storage/provider/src/bundle_state/bundle_state_with_receipts.rs

```diff
@@ -293,24 +293,42 @@ impl BundleStateWithReceipts {
         std::mem::swap(&mut self.bundle, &mut other)
     }
 
+    /// Write the [BundleStateWithReceipts] to database and receipts to either database or static
+    /// files, in append-only mode. See
+    /// [`write_to_storage_with_mode`](Self::write_to_storage_with_mode).
+    pub fn write_to_storage<TX>(
+        self,
+        tx: &TX,
+        static_file_producer: Option<StaticFileProviderRWRefMut<'_>>,
+        is_value_known: OriginalValuesKnown,
+    ) -> ProviderResult<()>
+    where
+        TX: DbTxMut + DbTx,
+    {
+        self.write_to_storage_with_mode(tx, static_file_producer, is_value_known, true)
+    }
+
     /// Write the [BundleStateWithReceipts] to database and receipts to either database or static
     /// files if `static_file_producer` is `Some`. It should be none if there is any kind of
     /// pruning/filtering over the receipts.
     ///
-    /// `omit_changed_check` should be set to true of bundle has some of it data
-    /// detached, This would make some original values not known.
-    pub fn write_to_storage<TX>(
+    /// `omit_changed_check` should be set to true if bundle has some of its data detached. This
+    /// would make some original values not known.
+    ///
+    /// Writes all changes in append-only mode, if `true` passed as corresponding parameter.
+    pub fn write_to_storage_with_mode<TX>(
         self,
         tx: &TX,
         mut static_file_producer: Option<StaticFileProviderRWRefMut<'_>>,
         is_value_known: OriginalValuesKnown,
+        append_only: bool,
```

Comment:
once we have the whole flow working - we should remove this and insert ETL pre state insertion

Reply:
yes definitely, will make an issue once this is merged so I can link to this code, ok?